homeassistant/util/executor.py

```diff
@@ -63,10 +63,13 @@ def join_or_interrupt_threads(
 class InterruptibleThreadPoolExecutor(ThreadPoolExecutor):
     """A ThreadPoolExecutor instance that will not deadlock on shutdown."""
 
-    def shutdown(self, *args: Any, **kwargs: Any) -> None:
+    def shutdown(
+        self, *args: Any, join_threads_or_timeout: bool = True, **kwargs: Any
```

Comment:
Why do we need the `join_threads_or_timeout` argument?

Reply:
I don't think we do.  Not sure if we ever did.

dropped in https://github.com/home-assistant/core/pull/117339/commits/8a9fe6a24a614d7fda2058248f653fa400a5a252

Reply:
`shutdown` gets called from https://github.com/python/cpython/blob/8edfa0b0b4ae4235bb3262d952c23e7581516d4f/Lib/asyncio/base_events.py#L606 via https://github.com/home-assistant/core/blob/47beddc6c6d3dc01de9544781c97fa5734bbc28b/homeassistant/runner.py#L194

We want database executor to gracefully close the connection before it can reach the code that would forcefully interrupt the threads that fail to join gracefully.  For the default executor we don't need to do that (current behavior) since there isn't a way to gracefully shut them down if they don't join on their own.

Reply:
OK, I see. Please amend the docstring to explain this, and also explain when calling `shutdown` with `join_threads_or_timeout=False` why we do that

Reply:
Adjusted the docstring and added some more comments